stdlib/string.gr

```diff
@@ -1867,30 +1867,58 @@ export let forEachCodePointi = (fn: (Number, Number) -> Void, str: String) => {
   void
 }
 
+@unsafe
 let trimString = (str: String, end: Bool) => {
-  let chars = explode(str), charsLength = length(str)
-  let mut i = 0, offset = 1
+  let (>>>) = WasmI32.shrU
+  let (+) = WasmI32.add
+  let (*) = WasmI32.mul
+  let (-) = WasmI32.sub
+  let (<) = WasmI32.ltU
+  let (==) = WasmI32.eq
+  let (!=) = WasmI32.ne
+
+  let mut stringPtr = WasmI32.fromGrain(str)
+  let charsLength = WasmI32.load(stringPtr, 4n)
+  stringPtr += 8n
+  let mut i = 0n, offset = 1n
   if (end) {
-    i = charsLength - 1
-    offset = -1
+    i = charsLength - 1n
+    offset = -1n
   }
-  for (; i < charsLength && i > -1; i += offset) {
-    let currentChar = chars[i]
+  let mut count = 0n
+  for (; i < charsLength; i += offset) {
+    // Get The Byte
+    let byte = WasmI32.load8U(stringPtr, i)
     // TODO: Use unicode whitespace property and unicode line terminator once github issue #661 is completed
+    // Check Multi-Byte Characters
+    if (byte == 0xEFn && WasmI32.load(stringPtr, i - 1n) >>> 8n == 0xBFBBEFn) {
+      i += offset * 2n
+      count += 3n
+      continue
```

Comment:
It took me a moment to wrap my head around why this was `continue` and not `break`. Could you add a comment?